src/raftstore/store/peer.rs

```diff
@@ -408,6 +408,14 @@ impl Peer {
         let mut ready = self.raft_group.ready();
         let is_applying = self.get_store().is_applying_snap();
         if is_applying {
+            if !raft::is_empty_snap(&ready.snapshot) {
+                if !self.get_store().is_canceling_snap() {
+                    warn!("receiving a new snap {:?} when applying the old one, try to abort.",
+                          ready.snapshot);
+                    self.mut_store().cancle_applying_snap();
```

Comment:
A hypothetical case:

Applying snapshot is slow for follower in a server heavy load. And the leader re-sends the same snapshot in a fast speed. What will happen then? Is there any chance(reasonable?) the follower could actually have enough time to apply the snapshot?